dotCMS/src/integration-test/java/com/dotcms/security/secret/ServiceIntegrationAPIImplTest.java

```diff
@@ -0,0 +1,301 @@
+        final User admin = TestUserUtils.getAdminUser();
+        final Host host = new SiteDataGen().nextPersisted();
+
+        //Let's create a set of secrets for a service
+        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
+        final ServiceSecrets secrets1 = builder1.withServiceKey("serviceKey-1-Host-1")
+                .withHiddenSecret("test:secret1", "secret-1")
+                .withHiddenSecret("test:secret2", "secret-2")
+                .withHiddenSecret("test:secret3", "secret3")
+                .withHiddenSecret("test:secret4", "secret-4")
+                .build();
+        //Save it
+        api.saveSecrets(secrets1, host, admin);
+
+        //Now we want to update one of the values within the secret.
+        //We want to change the value from `secret3` to `secret-3` for the secret named "test:secret3"
+        final Secret secret = Secret.newSecret("secret-3".toCharArray(), Type.STRING, false);
+        //Update the individual secret
+        api.saveSecret("serviceKey-1-Host-1", new Tuple2<>("test:secret3",secret), host, admin);
+        //The other properties of the object should remind the same so lets verify so.
+        final Optional<ServiceSecrets> serviceSecretsOptional = api.getSecretsForService("serviceKey-1-Host-1", host, admin);
+        assertTrue(serviceSecretsOptional.isPresent());
+        final ServiceSecrets recoveredBean = serviceSecretsOptional.get();
+        assertEquals("serviceKey-1-Host-1", recoveredBean.getServiceKey());
+
+        //We didn't modify the keys just the value associated with `test:secret3`
+        assertTrue(recoveredBean.getSecrets().containsKey("test:secret1"));
+        assertTrue(recoveredBean.getSecrets().containsKey("test:secret2"));
+        assertTrue(recoveredBean.getSecrets().containsKey("test:secret3"));
+        assertTrue(recoveredBean.getSecrets().containsKey("test:secret4"));
+
+        //now lets verify the values returned
+        assertEquals("secret-1", recoveredBean.getSecrets().get("test:secret1").getString());
+        assertEquals("secret-2", recoveredBean.getSecrets().get("test:secret2").getString());
+        assertEquals("secret-3", recoveredBean.getSecrets().get("test:secret3").getString());
+        assertEquals("secret-4", recoveredBean.getSecrets().get("test:secret4").getString());
+
+    }
+
+    @Test
+    public void Test_Save_Secrets_Then_Delete_Single_Secret_Entry_Then_Add_It_Again_With_New_Value_Then_Verify() throws DotDataException, DotSecurityException {
+        final ServiceIntegrationAPI api = APILocator.getServiceIntegrationAPI();
+        final User admin = TestUserUtils.getAdminUser();
+        final Host host = new SiteDataGen().nextPersisted();
+
+        //Let's create a set of secrets for a service
+        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
+        final ServiceSecrets secrets1 = builder1.withServiceKey("serviceKeyHost-1")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "serviceKeyHost-1" appears 7 times in this file; the first occurrence is on line 241](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)